NabtoEdgeClient/NabtoEdgeClient/MdnsScanner.swift

```diff
@@ -113,7 +114,7 @@ public class MdnsScanner: NSObject {
     private func apiEventCallback(ec: NabtoClientError) {
         if (ec == NABTO_CLIENT_EC_STOPPED) {
             // allow ARC to reclaim us
-            self.aliveSelf = nil
+            self.selfReference = nil
             return
         }
         guard (ec == NABTO_CLIENT_EC_OK) else {
```

Comment:
This callback comes when the future for `nabto_client_listener_new_mdns_result()` resolves. I cannot find documentation on the error codes it can resolve with (which we should probably also fix), but this code seems odd to me:
If STOPPED is returned we clear the selfReference
If OK is returned we handle it and arms the listener
If some other error we do not armListener, but we also do not let go of our selfReference.

Reply:
This sounds like a bug - and the same (wrong) pattern is also evident in the ConnectionEventListener. Should we just give up on all non-OK statuses (ie, remove self reference and not arm listener)?

Reply:
Since it is not documented it is hard to say. However, we should not "half-way-give-up" where we stop further actions but also keeps our self alive (unless I am missing some reason to stay alive). I would think if we get some non-OK status, we should only keep going if we know it is okay to do so. So my guess/intuition says: yes give up on all non-OK statuses